### PR TITLE
Accept any x25519 public key

### LINKED ISSUE #31: Accept non-canonical encodings of X25519 public keys
Currently `agreement::PublicKey` only implements `TryFrom<[u8;32]>` instead of `From`

This prevents using non-canonical curve elements. However, the standardised `x25519` function has a signature of  `x25519(k: [u8; 32], u: [u8; 32]) -> [u8; 32]` which doesn't fail and silently canonicalize the public type.

For example [`x25519-dalek` implements `From`](https://docs.rs/x25519-dalek/latest/x25519_dalek/struct.PublicKey.html#impl-From%3C%5Bu8%3B%2032%5D%3E).

### PR DESCRIPTION
**Pull request by nickray**
Fixes https://github.com/ycrypto/salty/issues/31.

### CODE REVIEW

**Comment by nickray** [comment]
A bit stumped why https://github.com/ycrypto/salty/actions/runs/5457535558/jobs/9931719708 is failing even after adding the schemas.

File: src/agreement.rs [hunk 1]
```diff
@@ -31,13 +25,12 @@ pub struct SecretKey(pub(crate) Scalar);
 // #[zeroize(drop)]
 pub struct SharedSecret(pub(crate) MontgomeryPoint);
 
-impl TryFrom<[u8; 32]> for PublicKey {
-    type Error = crate::Error;
+impl From<[u8; 32]> for PublicKey {
     /// Given a byte array, construct a x25519 `PublicKey`. It may fail if some
     /// underlying checks fail.
```
Comment (sosthene-nitrokey) [inline src/agreement.rs:30]: Outdated comment?

```suggestion
```

**Comment by sosthene-nitrokey** [comment]
using 

```rust
    #[test_wycheproof(
        "wycheproof/wycheproof-tests/tests/eddsa_test.json",
        "eddsa_verify_schema.json"
    )]
```

Fixes the error for me.

File: tests/wycheproof_x25519.rs [hunk 2]
```diff
@@ -17,24 +17,27 @@ mod wycheproof {
...
         let valid: bool;
 
-        if private.is_err() || public.is_err() || shared_expected.is_err() {
-            valid = false;
-        } else {
-            let public = agreement::PublicKey::try_from(public.unwrap());
-            if public.is_err() {
+        match (private, public, shared_expected) {
+            (Ok(private), Ok(public), Ok(shared_expected)) => {
+                let public = agreement::PublicKey::try_from(public);
```
Comment (sosthene-nitrokey) [inline tests/wycheproof_x25519.rs:22]: I think this should use `from` now ? There are a couple of other places where this should be the case.

**Comment by nickray** [comment]
> using
> 
> ```rust
>     #[test_wycheproof(
>         "wycheproof/wycheproof-tests/tests/eddsa_test.json",
>         "eddsa_verify_schema.json"
>     )]
> ```
> 
> Fixes the error for me.

Everything works locally, just not in CI, hence the stump.

**Pull request merged**